common/src/main/java/io/netty/util/HashedWheelTimer.java

```diff
@@ -347,6 +381,17 @@ public Timeout newTimeout(TimerTask task, long delay, TimeUnit unit) {
         if (unit == null) {
             throw new NullPointerException("unit");
         }
+        if (shouldLimitTimeouts()) {
+            long pendingTimeoutsCount = pendingTimeouts.get();
```

Comment:
this variable is not used in an atomic fashion and may not enforce the limits as intended. Consider instead always incrementing check if you overflowed, and then decrement if you exceeded the limit.